src/refspec.c

```diff
@@ -160,6 +160,30 @@ void git_refspec__free(git_refspec *refspec)
 	memset(refspec, 0x0, sizeof(git_refspec));
 }
 
+int git_refspec_parse(git_refspec **refspec, const char *input, int is_fetch)
+{
+	git_refspec *new_refspec;
+	assert(refspec && input);
+
```

Comment:
I'd like to make sure that we always sanitize the out parameter. So the first thing this function does should be `*refspec = NULL`.